challenges/cloud/sre.md

```diff
@@ -0,0 +1,384 @@
+intended, as well as to demonstrate your approach to automation.
+
+## Deployment
+
+Levels 3+ include a deployment step which should deploy the solution to a local
+Kubernetes cluster. The choice of which Kubernetes cluster is up to you, but 
+please choose one that has the ability to run on macOS and Linux.
+
+## Key Dependencies
+
+You are welcome to solve this challenge using tools you are most familiar with,
+but, at a minimum, please include the following:
+
+* Go
+* make
+* Docker
+
+## Level 1
+
+### Server
+
+* HTTP API to retrieve the replica count of the Kubernetes Deployment
+* One or two tests that cover happy and unhappy scenarios
+
+### Automation
+
+* Write a Dockerfile to build an image for the server
+* Ability to run the server
+
+## Level 2
+
+### Server
+
+* HTTP API to retrieve the replica count of the Kubernetes Deployment
+* HTTP API to set the replica count of the Kubernetes Deployment
+* One or two tests that cover happy and unhappy scenarios
+
+### Automation
+
+* Write a Dockerfile to build an image for the server
+* Ability to run the server
+* Ability to execute integration tests against the local Kubernetes cluster
+* One or two tests that cover happy and unhappy scenarios
+
+## Level 3
+
+### Server
+
+* HTTP API to retrieve the replica count of the Kubernetes Deployment
+* HTTP API to set the replica count of a the Kubernetes Deployment
+* HTTP API to get the list of available Deployments in the Kubernetes cluster
+* HTTP health check verifying Kubernetes connectivity
+* One or two tests that cover happy and unhappy scenarios
+
+### Automation
+
+* Write a Dockerfile to build an image for the server
+* Ability to run the server
+* Ability to execute integration tests against the local Kubernetes cluster
+* GitHub Actions for every commit that verifies code quality, ensures the software builds, and all tests pass
+  * Unit testing and static analysis is in scope
+  * Integration testing can be omitted/skipped
+
+### Deployment
+
+* Create a helm chart for the service that includes, a minimum: a Deployment, ServiceAccount and Service
+
+## Level 4
+
+### Design
+
+In addition to including details about the requirements below, the design
+document should include a section describing how this solution could be deployed
+to an AWS EKS cluster. Please note that you are not expected to implement any
+deployment automation for an EKS target, or create any AWS resources.
+
+### Server
+
+* HTTP API to retrieve the replica count of the Kubernetes Deployment
+* HTTP API to set the replica count of a the Kubernetes Deployment
+* HTTP API to get the list of available Deployments in the Kubernetes cluster
+* HTTP health check verifying Kubernetes connectivity
+* Extend the API to support a difference between current and desired state
+* Extend the Server to support the ability to show the differences between current and desired state (implementation of state storage for the server is left as a design choice for the candidate)
```

Comment:
What's the difference between both points?

Reply:
Yeah, I had the same question in the original version. We've updated it since but I agree that there is still too much overlap.